### ISSUE TITLE
Support mocking for async methods

### ISSUE THREAD

**Issue by sckott** [ropensci/crul#78]
Right now, mocking is only supported for `HttpClient`. Support it for if possible

- `Async`
- `AsyncVaried`

`AsyncVaried` is used inside of `Async`, so should just need to adjust `AsyncVaried`

**Comment by sckott** [comment]
see local branch `mock-async` 

It's not clear how to do this. 

I can't see a way to mock the mutli curl api, but maybe it can be dumber than that ...

**Comment by maelle** [comment]
ah, so is this a limit for http testing at the moment?

**Comment by sckott** [comment]
yes, right now async requests can't be tested with webmockr or vcr

**Comment by maelle** [comment]
Good to know!

**Comment by maelle** [comment]
so just as a note, in ropenaq tests that uses async requests got `skip_if_offline()` :-)
